### PR TITLE
fixes #6003 - don't render object directly

### PR DESCRIPTION
**Pull request by thomasmckay**
```
undefined method `key?' for #<JSON::Ext::Generator::State:0x0000000e782a68>
```

### CODE REVIEW

**Comment by dustints** [comment]
ack per ci

**Comment by domcleal** [comment]
Please address the failing tests:
Api::V2::UsersControllerTest.test_0004_shows default taxonomies on show response

File: app/views/api/v2/users/main.json.rabl [hunk 1]
```diff
@@ -5,10 +5,14 @@ extends "api/v2/users/base"
 attributes :firstname, :lastname, :mail, :admin, :auth_source_id, :auth_source_name, :last_login_on, :created_at, :updated_at
 
 if SETTINGS[:locations_enabled]
-  attributes :default_location
+  child :default_location, :root => :default_location do
+    attributes :id, :name
```
Comment (domcleal) [inline app/views/api/v2/users/main.json.rabl]: Should use an existing rabl

**Comment by isratrade** [comment]
@thomasmckay, this should work

```
child :default_location => :default_location do
  extends "api/v2/taxonomies/base"
end
```

**Comment by isratrade** [comment]
@thomasmckay, you can fix the test failure by 

``` ruby
-  assert_equal taxonomies(:location1).id, show_response['default_location']['location']['id']
+ assert_equal taxonomies(:location1).id, show_response['default_location']['id']
```

since the child node not doesn't have an object root.

**Comment by isratrade** [comment]
[test] seems like CI issue. only failure was 1.9.3 on mysql.

**Comment by domcleal** [comment]
Merged as ecac944 for Foreman 1.5.2, thanks @thomasmckay.

**Pull request closed without merging**